server/middlewares/user-right.ts

```diff
@@ -20,8 +20,34 @@ function ensureUserHasRight (userRight: UserRight) {
   }
 }
 
+async function ensureUserCanManageChannel (req: express.Request, res: express.Response, next: express.NextFunction) {
+  const user = res.locals.oauth.token.user
+  const isUserOwner = res.locals.videoChannel.Account.userId !== user.id
+
+  if (isUserOwner && user.hasRight(UserRight.MANAGE_ANY_VIDEO_CHANNEL) === false) {
+    const message = `User ${user.username} does not have right to manage channel ${req.params.nameWithHost}.`
+    logger.info(message)
+
+    return res.fail({
+      status: HttpStatusCode.FORBIDDEN_403,
+      message
+    })
+  }
+
+  const onUser = await res.locals.videoChannel.Account.$get('User')
+  if (user.role === UserRole.MODERATOR && onUser.role === UserRole.ADMINISTRATOR) {
```

Comment:
I don't think we need this.

IMHO a moderator should be trusted, and could update admin's channels.